ambry-router/src/main/java/com.github.ambry.router/PutManager.java

```diff
@@ -282,6 +283,11 @@ void onComplete(PutOperation op) {
       (op.isEncryptionEnabled() ? routerMetrics.putEncryptedBlobOperationLatencyMs
           : routerMetrics.putBlobOperationLatencyMs).update(operationLatencyMs);
     }
+    // Preemptively clean up of correlation IDs to avoid retaining a reference to the PutOperation for longer than
+    // required. We do not care about any other
+    for (Integer correlationId : op.getInFlightCorrelationIds()) {
+      correlationIdToPutOperation.remove(correlationId);
+    }
```

Comment:
Thanks for making this change. Can it be applied to GET Operation?

Reply:
I will look at this in the future for Get/Delete/TtlUpdate. The memory issues I saw in the heap dump did not seem to be as apparent for GetOperation since chunk buffer references are removed as soon as they are written out.